app.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+var express = require('express');
+var trucks = require('./trucks');
+var _ = require('underscore')
+var bodyParser = require('body-parser')
+var app = express();
+app.use(bodyParser.urlencoded({extended: true}));
+app.use(bodyParser.json());
+app.use(express.static('public'));
+
+app.get('/trucks', function (request, response) {
+  response.json(trucks.getTrucks());
+});
+
+app.post('/trucks', function (request, response){
+  var newTruck = request.body;
+  var foodTrucks = trucks.getTruck();
+  console.log(newTruck);
+  if (newTruck) {
+    var truckNames = _.pluck(foodTrucks, 'name')
+    var contains = _.contains(truckNames, newTruck.name)
+    if (!contains) {
+      trucks.addTruck(newTruck)
+      response.status(201).json('Truck added');
```

Comment:
After setting the status, the post handler should send the new truck to the response as well. This is why you were seeing undefined after adding a new truck, because the ajax done handler is constructing the markup to append to the list by using a truck object that should be passed in.